spacy/language.py

```diff
@@ -243,7 +243,8 @@ def config(self) -> Config:
         self._config["nlp"]["pipeline"] = list(self.component_names)
         self._config["nlp"]["disabled"] = list(self.disabled)
         self._config["components"] = pipeline
-        self._config["training"]["score_weights"] = combine_score_weights(score_weights)
+        if not self._config["training"].get("score_weights"):
```

Comment:
Alternatively, I guess we could use `setdefault` here? Then we don't need the extra `get`?

Reply:
Erm, yea I guess we could? It feels a bit funny to use `setdefault` with an elaborate, inferred list of things that is different everytime this function is run though? Not sure it'll be as readable.

Reply:
And come to think about it, it can't be more efficient? At some point, the internal dict needs to check whether it needs to use the default value anyway...

Reply:
Yeah we shouldn't worry about efficiency, it won't be significant here... sorry, I think my suggestion was kinda dumb, it really doesn't matter and it's not going to be more readable.

(Also, every time I use `setdefault` I have to double-check the syntax 😂 So that's already a bad sign.)